### ISSUE TITLE
Taskbar style  "Windows 11". Explorer crashes upon right-clicking the taskbar

### ISSUE THREAD

**Issue by kevwkev** [valinet/ExplorerPatcher#1805]
After updating, when the Taskbar style is configured as "Windows 11 (default)", the Explorer crashes or restarts upon right-clicking the taskbar. This problem is observed in EP Version 22621.1992.56.1 with OS Build 10.0.22621.2134.

**Comment by lenovoscmt** [comment]
Same from here. Buggy update.

**Comment by DocmoPS** [comment]
yes, mine crashing on load (after rebooting for me).  tries a handful of times and then gives the error prompt which led me here.

EXCELLENT APP though.  A huge productivity saver.  Thanks again from a happy supporter.

**Comment by aberaiden** [comment]
Same as #1802 ?
Remove IsUndockedAssetAvailable in HKEY_CURRENT_USER\Software\ExplorerPatcher\twinui.pcshell (or change the value to 36c78) as a temporary fix. This will stop the crashing, but Win+Tab is not working.

**Comment by MentzWolf** [comment]
Can confirm I'm having this issue as well, guessing it has something to do with the latest Windows update?

**Comment by ItsMasterL** [comment]
I was also having this issue, but it seems to work fine after just restarting it a few times. Still annoying nonetheless

**Comment by fuste76** [comment]
Win+Tab is not working on Windows 10 Taskbar style, too.

**Comment by Sakata-MC** [comment]
> Same as #1802 ? Remove IsUndockedAssetAvailable in HKEY_CURRENT_USER\Software\ExplorerPatcher\twinui.pcshell (or change the value to 36c78) as a temporary fix. This will stop the crashing, but Win+Tab is not working.

These fixes unfortunately aren't working for me.

**Comment by slapfightchampionship** [comment]
Same for me, started happening after latest windows update. Hope this gets fixed soon!!

**Comment by slapfightchampionship** [comment]
> 

restarting what several times? windows or reinstall or?

**Comment by honzawashere** [comment]
Same issue for me. After right clicking the taskbar, it should open the context menu. Instead of it, it just crashes. I though that TranslucentTB or Wallpaper Engine has something with it, but unfortunately it's caused by ExplorerPatcher, after uninstalling everything works well again. But I miss my Windows 10 taskbar.

**Comment by Girofox** [comment]
<img width="485" alt="image" src="https://github.com/valinet/ExplorerPatcher/assets/43116061/78008dc5-8ce5-4f21-97dc-fb4cfab9f030">

To fix the right click crashing you need to go into `EP settings` via Search and tick ✔️ `Do not change the taskbar context menu`

The registry trick alone under #1802 did not worked for me.

**Comment by wefeelalive** [comment]
identical problem, and upon an update, it communicates with a new server it seems. 
https://www.virustotal.com/gui/file/55fccc26a7b473a2bd91d59c126c3bb952c51430516523d62f700fad7ed292e7/relations 
Can someone explain why that IP is marked as having 8/88 positive virus detections?

**Comment by slapfightchampionship** [comment]
> <img alt="image" width="485" src="https://user-images.githubusercontent.com/43116061/259553867-78008dc5-8ce5-4f21-97dc-fb4cfab9f030.png">
> To fix the right click crashing you need to go into `EP settings` via Search and tick ✔️ `Do not change the taskbar context menu`
> 
> The registry trick alone under #1802 did not worked for me.

How do you get into EP settings via Search???

**Comment by wefeelalive** [comment]
> > <img alt="image" width="485" src="https://user-images.githubusercontent.com/43116061/259553867-78008dc5-8ce5-4f21-97dc-fb4cfab9f030.png">
> > To fix the right click crashing you need to go into `EP settings` via Search and tick ✔️ `Do not change the taskbar context menu`
> > The registry trick alone under #1802 did not worked for me.
> 
> How do you get into EP settings via Search???

You search in the windows search bar for Properties or in your case Eigenschaften and then press on the ExploererPatcher Properties and there you go. But for me that fix ain't working out.

**Comment by unitmotion** [comment]
<img width="266" alt="image" src="https://github.com/valinet/ExplorerPatcher/assets/34281821/06735be8-d237-4e85-8187-486567578f5b">
<img width="175" alt="image" src="https://github.com/valinet/ExplorerPatcher/assets/34281821/8515d2a3-fdf8-439b-af4c-bc7574d1fc22">

**Comment by slapfightchampionship** [comment]
> <img alt="image" width="266" src="https://user-images.githubusercontent.com/34281821/259562981-06735be8-d237-4e85-8187-486567578f5b.png"> <img alt="image" width="175" src="https://user-images.githubusercontent.com/34281821/259563030-8515d2a3-fdf8-439b-af4c-bc7574d1fc22.png"> but it didn't help.

Thanks it worked!

**Comment by lokpro** [comment]
using my settings, crashes immediately when open. No crashes after cleared the Reg.
<details> 
```reg
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\ExplorerPatcherx]
"MigratedFromOldSettings"=dword:00000001
"FileExplorerCommandUI"=dword:00000000
"OrbStyle"=dword:00000001
"SkinMenus"=dword:00000001
"CenterMenus"=dword:00000001
"FlyoutMenus"=dword:00000001
"HideControlCenterButton"=dword:00000000
"HideExplorerSearchBar"=dword:00000000
"ShrinkExplorerAddressBar"=dword:00000000
"UseClassicDriveGrouping"=dword:00000000
"StartDocked_DisableRecommendedSection"=dword:00000001
"DisableOfficeHotkeys"=dword:00000000
"TaskbarAutohideOnDoubleClick"=dword:00000000
"ToolbarSeparators"=dword:00000000
"OSBuild"="10.0.22621.1992"
"DisableAeroSnapQuadrants"=dword:00000001
"OpenPropertiesAtNextStart"=dword:00000000
"IsUpdatePending"=dword:00000000
"CrashCounter"=dword:00000000
"OldTaskbar"=dword:00000000

[HKEY_CURRENT_USER\Software\ExplorerPatcherx\sws]

[HKEY_CURRENT_USER\Software\ExplorerPatcherx\twinui.pcshell]
"CImmersiveContextMenuOwnerDrawHelper::s_ContextMenuWndProc"=dword:002489e0
"CLauncherTipContextMenu::GetMenuItemsAsync"=dword:0064b900
"ImmersiveContextMenuHelper::ApplyOwnerDrawToMenu"=dword:0069f3dc
"ImmersiveContextMenuHelper::RemoveOwnerDrawFromMenu"=dword:0069fbe4
"CLauncherTipContextMenu::_ExecuteShutdownCommand"=dword:0065dc2c
"CLauncherTipContextMenu::_ExecuteCommand"=dword:0065d8f4
"CLauncherTipContextMenu::ShowLauncherTipContextMenu"=dword:0064cf50
"IsUndockedAssetAvailable"=dword:00026f90
"winrt::Windows::Internal::Shell::implementation::MeetAndChatManager::OnMessage"=dword:004d61cc

```
</details>


Edition	Windows 11 Pro
Version	22H2
Installed on	‎7/‎11/‎2022
OS build	22621.2134
Experience	Windows Feature Experience Pack 1000.22659.1000.0

**Comment by AurumXion** [comment]
> After updating, when the Taskbar style is configured as "Windows 11 (default)", the Explorer crashes or restarts upon right-clicking the taskbar. This problem is observed in EP Version 22621.1992.56.1 with OS Build 10.0.22621.2134.



> After updating, when the Taskbar style is configured as "Windows 11 (default)", the Explorer crashes or restarts upon right-clicking the taskbar. This problem is observed in EP Version 22621.1992.56.1 with OS Build 10.0.22621.2134.

i figured it out guys. just go to the properties (ExplorerPatcher) > Advanced > Do not change the taskbar context menu (e.g . do not display the "properties" item)

**Comment by Eaglesg** [comment]
> <img alt="image" width="485" src="https://user-images.githubusercontent.com/43116061/259553867-78008dc5-8ce5-4f21-97dc-fb4cfab9f030.png">
> 
> To fix the right click crashing you need to go into `EP settings` via Search and tick ✔️ `Do not change the taskbar context menu`
> 
> The registry trick alone under #1802 did not worked for me.

Same issue for me (EP Version 22621.1992.56.1 with OS Build 10.0.22621.2134) and this temporary fix is working.

**Comment by Bookmakr** [comment]
The change in Advanced Properties solved the issue for me, but only after rebooting. Before, Explorer continued to restart.
After the restart, there were also some symbol data installed (I missed, if from Explorer or from ExplorerPatcher.

By the way: **Great application - good to know, that some people understand, what Microsoft doesn't! Thank you so much.**
(I'll rate it 'Best Tool of the decade', together with some other like OpenShell etc)

**Comment by wefeelalive** [comment]
> The change in Advanced Properties solved the issue for me, but only after rebooting. Before, Explorer continued to restart. After the restart, there were also some symbol data installed (I missed, if from Explorer or from ExplorerPatcher.
> 
> By the way: **Great application - good to know, that some people understand, what Microsoft doesn't! Thank you so much.** (I'll rate it 'Best Tool of the decade', together with some other like OpenShell etc)

Even if I don't do anything, it keeps crashing even after rebooting and doing the fix.

**Comment by wefeelalive** [comment]
> it

 deleting IsUndockedAssetAvailable (located at HKEY_CURRENT_USER\Software\ExplorerPatcher\twinui.pcshell) and OSBuild (located at HKEY_CURRENT_USER\Software\ExplorerPatcher) helped

**Comment by aldeniangman** [comment]
> Same as #1802 ? Remove IsUndockedAssetAvailable in HKEY_CURRENT_USER\Software\ExplorerPatcher\twinui.pcshell (or change the value to 36c78) as a temporary fix. This will stop the crashing, but Win+Tab is not working.

Did not work for me when just changed it to 36c78
WAITING for next update

**Comment by aldeniangman** [comment]
looks good so far

**Comment by aldeniangman** [comment]
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOo

**Comment by wefeelalive** [comment]
> > Same as #1802 ? Remove IsUndockedAssetAvailable in HKEY_CURRENT_USER\Software\ExplorerPatcher\twinui.pcshell (or change the value to 36c78) as a temporary fix. This will stop the crashing, but Win+Tab is not working.
> 
> Did not work for me when just changed it to 36c78 WAITING for next update

Remove IsUndockedAssetAvailable and don't change the Value and it should work out!

**Comment by pellajl** [comment]
I had the same issue but sorted it out:

1. Uninstalled ExplorerPatcher
2. Deleted all references to ExplorerPatcher in the registry.
3. Deleted C:\Program Files\ExplorerPatcher.
4. Reinstalled ExplorerPatcher.

I hope this is a permanent fix and doesn't occur again.

**Comment by GinoMan** [comment]
> I had the same issue but sorted it out:
> 
>     1. Uninstalled ExplorerPatcher
> 
>     2. Deleted all references to ExplorerPatcher in the registry.
> 
>     3. Deleted C:\Program Files\ExplorerPatcher.
> 
>     4. Reinstalled ExplorerPatcher.
> 
> 
> I hope this is a permanent fix and doesn't occur again.

This is a workaround. Fix is we install a newer version of explorer patcher and the issue goes away when we enable the context menu hook.

**Comment by e-t-l** [comment]
> I had the same issue but sorted it out:
> 
>     1. Uninstalled ExplorerPatcher
> 
>     2. Deleted all references to ExplorerPatcher in the registry.
> 
>     3. Deleted C:\Program Files\ExplorerPatcher.
> 
>     4. Reinstalled ExplorerPatcher.
> 

I did this exactly and the issue still persisted. I tried deleting the two registry keys and the issue still persisted. I'm at a bit of a loss, because the taskbar context menu was extremely helpful, especially compared to the crummy Win 11 one.

By the way, has anyone noticed that similarly, "doubleclick taskbar to toggle autohide" has stopped working? I have it enabled and yet doubleclicking the taskbar doesn't change its autohide behavior. No idea if this could possibly be related to the context menu bug.

**Comment by pyrates999** [comment]
Possible fix: In ExplorerPatcher Properties, go to Advanced and tick "Do not change the taskbar context menu"

**Comment by e-t-l** [comment]
> Possible fix: In ExplorerPatcher Properties, go to Advanced and tick "Do not change the taskbar context menu"

Thank you, I should have specified that I am currently using that workaround. But only because the uninstall-reboot-install and it didn't help.

**Comment by Sakata-MC** [comment]
Same here. Having to use workaround.
I'm hoping devs have some ideas of what's going on and can fix. The connect
menu change was one of the features I used.

**Comment by Edwin-Baranov** [comment]
> I had the same issue but sorted it out:
> 
> 1. Uninstalled ExplorerPatcher
> 2. Deleted all references to ExplorerPatcher in the registry.
> 3. Deleted C:\Program Files\ExplorerPatcher.
> 4. Reinstalled ExplorerPatcher.
> 
> I hope this is a permanent fix and doesn't occur again.

I did everything you mentioned and it still crashes

**Comment by pyrates999** [comment]
duplicate of: https://github.com/valinet/ExplorerPatcher/issues/2051

Please close this one.

Possible fixes:

1. In ExplorerPatcher Properties, go to Advanced and tick **Do not change the taskbar context menu**
2. In ExplorerPatcher Properties, go to File Explorer and untick **Disable the windows 11 context menu**
3. In ExplorerPatcher Properties, go to Taskbar and set **Taskbar style** to **Windows 11**

**Comment by aldeniangman** [comment]
No crashing problem for me.
________________________________

**Comment by Ringlings** [comment]
> duplicate of: #2051

It's not? The issue you linked is a bootloop, this is a crash on right click. And this issue is older than the one you linked anyway.

**Comment by pyrates999** [comment]
> > duplicate of: #2051
> 
> It's not? The issue you linked is a bootloop, this is a crash on right click. And this issue is older than the one you linked anyway.

This is now for a previous KB.  So it is no longer relevant.

**Comment by DavidBradbury** [comment]
> duplicate of: #2051
> 
> Please close this one.
> 
> Possible fixes:
> 
>     1. In ExplorerPatcher Properties, go to Advanced and tick **Do not change the taskbar context menu**
> 
>     2. In ExplorerPatcher Properties, go to File Explorer and untick **Disable the windows 11 context menu**
> 
>     3. In ExplorerPatcher Properties, go to Taskbar and set **Taskbar style** to **Windows 11**

Traditionally, the older issue is kept and the newer one is labeled the duplicate. This is still happening in the latest release, 22621.1992.56.3 - Also, those are not fixes, those are workarounds.

**Comment by pyrates999** [comment]
> > duplicate of: #2051
> > Please close this one.
> > Possible fixes:
> > ```
> > 1. In ExplorerPatcher Properties, go to Advanced and tick **Do not change the taskbar context menu**
> > 
> > 2. In ExplorerPatcher Properties, go to File Explorer and untick **Disable the windows 11 context menu**
> > 
> > 3. In ExplorerPatcher Properties, go to Taskbar and set **Taskbar style** to **Windows 11**
> > ```
> 
> Traditionally, the older issue is kept and the newer one is labeled the duplicate. This is still happening in the latest release, 22621.1992.56.3 - Also, those are not fixes, those are workarounds.

words words words.

**Comment by Girofox** [comment]
So far the random crahsea stopped for me all together, even with the lastest cumulative update from mid September

**Comment by Amrsatrio** [comment]
Please install the newly released EP build. There should be no more major issues on 22621 builds up to 22621.2359.
